pkg/operation/botanist/component/logging/fluentoperator/custom_resources.go

```diff
@@ -0,0 +1,210 @@
+	additionalFilters []*fluentbitv1alpha2.ClusterFilter,
+	additionalParsers []*fluentbitv1alpha2.ClusterParser,
+) component.DeployWaiter {
+	return &customResources{
+		client:            client,
+		namespace:         namespace,
+		values:            values,
+		additionalInputs:  additionalInputs,
+		additionalFilters: additionalFilters,
+		additionalParsers: additionalParsers,
+	}
+}
+
+func (c *customResources) Deploy(ctx context.Context) error {
+	var (
+		registry = managedresources.NewRegistry(kubernetes.SeedScheme, kubernetes.SeedCodec, kubernetes.SeedSerializer)
+
+		config = &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      v1beta1constants.DaemonSetNameFluentBit + "-lua-config",
+				Namespace: c.namespace,
+			},
+			Data: map[string]string{
+				"modify_severity.lua": `
+function cb_modify(tag, timestamp, record)
+  local unified_severity = cb_modify_unify_severity(record)
+
+  if not unified_severity then
+    return 0, 0, 0
+  end
+
+  return 1, timestamp, record
+end
+
+function cb_modify_unify_severity(record)
+  local modified = false
+  local severity = record["severity"]
+  if severity == nil or severity == "" then
+	return modified
+  end
+
+  severity = trim(severity):upper()
+
+  if severity == "I" or severity == "INF" or severity == "INFO" then
+    record["severity"] = "INFO"
+    modified = true
+  elseif severity == "W" or severity == "WRN" or severity == "WARN" or severity == "WARNING" then
+    record["severity"] = "WARN"
+    modified = true
+  elseif severity == "E" or severity == "ERR" or severity == "ERROR" or severity == "EROR" then
+    record["severity"] = "ERR"
+    modified = true
+  elseif severity == "D" or severity == "DBG" or severity == "DEBUG" then
+    record["severity"] = "DBG"
+    modified = true
+  elseif severity == "N" or severity == "NOTICE" then
+    record["severity"] = "NOTICE"
+    modified = true
+  elseif severity == "F" or severity == "FATAL" then
+    record["severity"] = "FATAL"
+    modified = true
+  end
+
+  return modified
+end
+
+function trim(s)
+  return (s:gsub("^%s*(.-)%s*$", "%1"))
+end`,
+				"add_tag_to_record.lua": `
+function add_tag_to_record(tag, timestamp, record)
+  record["tag"] = tag
+  return 1, timestamp, record
+end`,
+			},
+		}
+	)
+
+	utilruntime.Must(kubernetesutils.MakeUnique(config))
+
+	resources := []client.Object{config}
+
+	fluentBit := customresources.GetFluentBit(getFluentBitLabels(), v1beta1constants.DaemonSetNameFluentBit, c.namespace, c.values.FluentBitImage, c.values.FluentBitInitImage, c.values.FluentBitPriorityClass)
+	resources = append(resources, fluentBit)
+
+	clusterFluentBitConfig := customresources.GetClusterFluentBitConfig(v1beta1constants.DaemonSetNameFluentBit, getCustomResourcesLabels())
+	resources = append(resources, clusterFluentBitConfig)
+
+	for _, clusterInput := range customresources.GetClusterInputs(getCustomResourcesLabels()) {
+		resources = append(resources, clusterInput)
+	}
```

Comment:
This should work, so you can get rid of the `for` loops this way:

```suggestion
	resources = append(resources, customresources.GetClusterInputs(getCustomResourcesLabels())...)
```

Reply:
```
cannot use customresources.GetClusterInputs(getCustomResourcesLabels()) (value of type []*v1alpha2.ClusterInput) as []client.Object value in argument to append
```

`GetClusterInputs` function returns a slice with pointers